doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
+
+Module Repositories
+*******************
+
+* All modules included in the default manifest shall be hosted in repositories
+  under the Zephyr Project Github organization.
+
+* Module repository names should follow the convention of using lowercase
+  letters and dashes instead of underscores. This rule will apply to all
+  new module repositories.
+
+  .. note::
+
+     It is not required to modify the names of existing module repositories
+     to make them comply with the above naming convention.
+
+* If the module has an external (upstream) project repository, the module
+  repository should preserve the upstream repository folder structure.
+
+* Modules should use "zephyr" as the default name for the repository main
+  branch. Branches for specific purposes, for example, a module branch for
+  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+
+  .. note::
+
+     It is **not** recommended in module repositories to have a 'master'
+     branch mirroring the master branch of the external repository.
+
+.. _modules_synchronization:
+
+Synchronizing with upstream
+===========================
+
+Requirements for allowed practices
+----------------------------------
+
+Any changes to module repositories, as a result of synchronizing with,
+upstream shall be brought in by pull requests. These pull requests
+shall be merg-able (i.e. with Rebase & Merge option, using Github UI).
+This ensures that a linear history is preserved in the module repositories.
+It also allows to easily review the incoming changea and run Zephyr CI,
+git lint, identity, and license checks directly on the set of changes
+that are to be brought into the module.
+
+.. note::
+
+     Force-pushing to a module's main branch is not allowed.
+
+Allowed practices
+---------------------
+
+The following practices are allowed and should be followed in all modules.
+
+**Updating modules with a diff from upstream:**
+Upstream changes brought as a single commit (manual diff) in a PR against
+the module's main branch. This approach is simple and should be applicable
+to all modules with the downside of supressing the upstream history in the
+module repository.
+
+The commit message is expected to identify the upstream repository, the
+version to which the module is updated (upstream version, tag, commit,
+etc.), and the reason for the doing the update.
+
+**Cherry picking upstream commits:**
+Upstream changes brought in by cherry-picking the upstream commits and
+submitting a PR with these commits against the module main branch. This
+approach is applicable to modules with an upstream project Git repository.
+The main advantages of this approach are the following
+
+* the upstream repository history may be preserved in the module repository,
+  if the cherry picked commits have a pointer to the original upstream commit
+* downstream module patches are less likely to be unintentionally overwritten
+  by the synchronization process.
+
```

Comment:
Why not have ** merge upstera commits ** option?
It is very simple to be used - just need to ensure descriptive message in the merge commit.
Such synchronization PR is always merged by push to the repo.
So fare I have been using this method for synchronize mcuboot repo.

Reply:
to be discussed.

Reply:
I agree, I've used this method in OpenThread module so far as well. This allows to retain the upstream OT repo history and make synchronization points (merge commits) explicit.